rclcpp/include/rclcpp/any_subscription_callback.hpp

```diff
@@ -35,17 +35,23 @@ struct AnySubscriptionCallback
   using SharedPtrCallback = std::function<void(const std::shared_ptr<MessageT> &)>;
   using SharedPtrWithInfoCallback =
       std::function<void(const std::shared_ptr<MessageT> &, const rmw_message_info_t &)>;
+  using ConstSharedPtrCallback = std::function<void(const std::shared_ptr<const MessageT> &)>;
```

Comment:
I will remove references to shared_ptrs but not to unique_ptrs as this seems to mess up how unique_ptrs are being copied for intra-process.